includes/data-stores/class-wc-customer-data-store.php

```diff
@@ -324,22 +325,31 @@ private function update_user_meta( $customer ) {
 	 * @return WC_Order|false
 	 */
 	public function get_last_order( &$customer ) {
-		global $wpdb;
-
-		$last_order = $wpdb->get_var(
-			// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
-			"SELECT posts.ID
-			FROM $wpdb->posts AS posts
-			LEFT JOIN {$wpdb->postmeta} AS meta on posts.ID = meta.post_id
-			WHERE meta.meta_key = '_customer_user'
-			AND   meta.meta_value = '" . esc_sql( $customer->get_id() ) . "'
-			AND   posts.post_type = 'shop_order'
-			AND   posts.post_status IN ( '" . implode( "','", array_map( 'esc_sql', array_keys( wc_get_order_statuses() ) ) ) . "' )
-			ORDER BY posts.ID DESC"
-			// phpcs:enable
+		$last_order = apply_filters(
+			'woocommerce_customer_get_last_order',
+			get_user_meta( $customer->get_id(), '_last_order', true ),
+			$customer
 		);
 
-		if ( ! $last_order ) {
+		if ( '' === $last_order ) {
+			global $wpdb;
+
+			$last_order = $wpdb->get_var(
+				// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+				"SELECT posts.ID
+				FROM $wpdb->posts AS posts
+				LEFT JOIN {$wpdb->postmeta} AS meta on posts.ID = meta.post_id
+				WHERE meta.meta_key = '_customer_user'
+				AND   meta.meta_value = '" . esc_sql( $customer->get_id() ) . "'
+				AND   posts.post_type = 'shop_order'
+				AND   posts.post_status IN ( '" . implode( "','", array_map( 'esc_sql', array_keys( wc_get_order_statuses() ) ) ) . "' )
+				ORDER BY posts.ID DESC"
+				// phpcs:enable
+			);
+			update_user_meta( $customer->get_id(), '_last_order', $last_order );
+		}
+
+		if ( ! $last_order || '' === $last_order ) {
```

Comment:
The second condition check isn't necessary because `! $last_order` evaluates to true when `$last_order` is an empty string.

Reply:
ouch, sure, sorry. Removing.